net/http/httperror/httperror.go

```diff
@@ -0,0 +1,101 @@
+// Package httperror defines the format for HTTP error responses
+// from Chain services.
+package httperror
+
+import (
+	"context"
+	"net/http"
+
+	"chain/errors"
+	"chain/log"
+	"chain/net/http/httpjson"
+	"chain/net/http/reqid"
+)
+
+func init() {
+	log.SkipFunc("chain/net/http/httperror.Formatter.Log")
+	log.SkipFunc("chain/net/http/httperror.Formatter.Write")
+}
+
+// Info contains a set of error codes to send to the user.
+type Info struct {
+	HTTPStatus int    `json:"-"`
+	ChainCode  string `json:"code"`
+	Message    string `json:"message"`
+}
+
+// Response defines the error response for a Chain error.
+type Response struct {
+	Info
+	Detail    string                 `json:"detail,omitempty"`
+	Data      map[string]interface{} `json:"data,omitempty"`
+	Temporary bool                   `json:"temporary"`
+}
+
+// Formatter defines rules for mapping errors to the Chain error
+// response format.
+type Formatter struct {
+	Default     Info
+	IsTemporary func(info Info, err error) bool
+	Errors      map[error]Info
+}
+
+// Format builds an error Response body describing err by consulting
+// the f.Errors lookup table. If no entry is found, it returns f.Default.
+func (f Formatter) Format(err error) (body Response, info Info) {
```

Comment:
Why return `info` separately from `body`, if it's also always available as `body.Info`? Is it just a convenience? Does it conform to some interface?